### PR TITLE
Ensure OU entries are not reordered.

### LINKED ISSUE #6596: Vault PKI shouldn't sort OUs
**Describe the bug**
Vault PKI alphabetically sorts all OUs when multiple ones are present in a requested CSR.

**To Reproduce**
1. Enable the pki engine: `vault secrets enable pki`
2. Generate the CSR for an intermediate CA: `vault write -field=csr pki/intermediate/generate/internal country="US" organization="Hashicorp" ou="C,A,B" common_name="My CA" key_type="rsa" key_bits=4096 add_basic_constraints=true > My_CA.csr`
3. Show the subject of the generated CSR: `openssl req -in My_CA.csr -noout -subject`
4. See that the three OUs will be ordered like `OU = A + OU = B + OU = C`

**Expected behavior**
Because the RDNs impose a hierarchy, the OUs should follow the order that was defined in the request, i.e. they should be ordered (in this example) like `OU = C + OU = A + OU = B`.

You can further inspect the CSR and verify that the OUs were actually ordered by running `openssl asn1parse -i -in My_CA.csr`.

**Environment:**
* Vault Server Version: 1.1.1
* Vault CLI Version: 1.1.1
* Server Operating System/Architecture: GNU/Linux x86_64 (Ubuntu Server 18.04.2 LTS)

Vault server configuration file(s):

```hcl
storage "file" {
  path = "/opt/vault/storage"
}
listener "tcp" {
  address = "127.0.0.1:8200"
  tls_disable = true
}
api_addr = "http://127.0.0.1:8200"
pid_file = "/opt/vault/vault.pid"
ui = true
raw_storage_endpoint = true
```

**Additional context**
https://en.wikipedia.org/wiki/Directory_information_tree
https://tools.ietf.org/html/rfc5280#section-4.1.2.6

### PR DESCRIPTION
**Pull request by mgritter**
Closes #6596.

This is more minimal than I would really like, but it resolves the issue somebody actually complained about without opening too big a can of worms.

It's questionable whether we should be deduplicating this field at all; the other fields probably should be treated the same way.

And I'm not happy that x509 writes out a _set_ of OUs in the CSR, but then parses it into a _sequence_ of OUs when reading it back in, but that doesn't have an easy fix.

### CODE REVIEW

**Comment by mgritter** [comment]
Passes original use case:

mgritter@ubuntu:~/vault$ vault secrets enable pki
Success! Enabled the pki secrets engine at: pki/
mgritter@ubuntu:~/vault$ vault write -field=csr pki/intermediate/generate/internal country="US" organization="Hashicorp" ou="C,A,B" common_name="My CA" key_type="rsa" key_bits=4096 add_basic_constraints=true > My_CA.csr
mgritter@ubuntu:~/vault$ openssl req -noout -in My_CA.csr -subject
subject=C = US, O = Hashicorp, OU = C + OU = A + OU = B, CN = My CA

File: builtin/logical/pki/cert_util_test.go [hunk 1]
```diff
@@ -123,3 +125,36 @@ func TestPki_FetchCertBySerial(t *testing.T) {
 		}
 	}
 }
+
+// Demonstrate  that multiple OUs in the name are handled in an
```
Comment (kalafut) [inline builtin/logical/pki/cert_util_test.go]:
```suggestion
// Demonstrate that multiple OUs in the name are handled in an
```

File: builtin/logical/pki/cert_util_test.go [hunk 2]
```diff
@@ -123,3 +125,36 @@ func TestPki_FetchCertBySerial(t *testing.T) {
...
+// order-preserving way.
+func TestPki_MultipleOUs(t *testing.T) {
+	var b backend;
```
Comment (kalafut) [inline builtin/logical/pki/cert_util_test.go]:
```suggestion
	var b backend
```

File: sdk/helper/strutil/strutil.go [hunk 3]
```diff
@@ -239,6 +239,28 @@ func RemoveDuplicates(items []string, lowercase bool) []string {
 	return items
 }
 
+// RemoveDuplicatesStable removes duplicate and empty elements from a slice of
+// strings, preserving order (and case) of the original slice.
+// In all cases, strings are compared after trimming whitespace
+// If caseInsensitive, strings will be compared after ToLower()
+func RemoveDuplicatesStable(items []string, caseInsensitive bool) []string {
+	itemsMap := map[string]bool{}
```
Comment (kalafut) [inline sdk/helper/strutil/strutil.go]:
```suggestion
	itemsMap := make(map[string]bool, len(items))
```

File: sdk/helper/strutil/strutil.go [hunk 4]
```diff
@@ -239,6 +239,28 @@ func RemoveDuplicates(items []string, lowercase bool) []string {
...
+	deduplicated := make([]string, 0, len(itemsMap))
```
Comment (kalafut) [inline sdk/helper/strutil/strutil.go]:
```suggestion
	deduplicated := make([]string, 0, len(items))
```

File: sdk/helper/strutil/strutil.go [hunk 5]
```diff
@@ -239,6 +239,28 @@ func RemoveDuplicates(items []string, lowercase bool) []string {
...
+	for _, item := range items {
+		key := strings.TrimSpace(item)
+		if caseInsensitive {
+			key = strings.ToLower(key)
+		}
+		if key == "" || itemsMap[key] {
+			continue
+		}
+		itemsMap[key] = true
+		deduplicated = append( deduplicated, item )
```
Comment (kalafut) [inline sdk/helper/strutil/strutil.go]: Odd spacing around the parens. Please check gofmt.

File: sdk/helper/strutil/strutil.go (hunk 4 above)
Comment (mgritter) [reply sdk/helper/strutil/strutil.go]: oops, good catch.

File: builtin/logical/pki/cert_util_test.go [hunk 6]
```diff
@@ -123,3 +125,36 @@ func TestPki_FetchCertBySerial(t *testing.T) {
...
+
+	apiData := &framework.FieldData{
+		Schema: fields,
+		Raw: map[string]interface{} {
+			"cn" : "example.com",
+			"ttl" : 3600,
+		},
+	}
+	input := &dataBundle{
+		apiData : apiData,
```
Comment (kalafut) [inline builtin/logical/pki/cert_util_test.go]: Odd spacing around the punctuation (here and elsewhere). Please check gofmt.

Comment (mgritter) [reply builtin/logical/pki/cert_util_test.go]: will do

**Comment by kalafut** [comment]
Just a few minor things otherwise LGTM!

**Pull request merged**